ENIGMAsystem/SHELL/Platforms/Win32/WINDOWSmain.cpp

```diff
@@ -463,12 +463,19 @@ void execute_program(std::string fname, std::string args, bool wait) { execute_p
 
 // converts a relative path to absolute if the path exists
 std::string filename_absolute(std::string fname) {
-  if (file_exists(fname) || directory_exists(fname)) {
-    wchar_t rpath[MAX_PATH];
-    tstring tstr_fname = widen(fname);
-    GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL); 
-    return shorten(rpath);
-  } else { return ""; }
+  if (string_replace_all(fname, " ", "") == "") fname = ".";
+  wchar_t rpath[MAX_PATH];
+  tstring tstr_fname = widen(fname);
+  GetFullPathNameW(tstr_fname.c_str(), MAX_PATH, rpath, NULL);
+  if (directory_exists(shorten(rpath))) return add_slash(shorten(rpath));
+  if (file_exists(shorten(rpath))) return shorten(rpath);
+  return "";
+}
+
+std::string filename_join(std::string prefix, std::string suffix) {
+  if (directory_exists(prefix))
```

Comment:
Let's avoid this method doing I/O. The typical use case will be to pass the result of this operation directly to another function, so returning the empty string won't help.

Reply:
fixed